src/client/pydaos/pydaos_core.py

```diff
@@ -69,20 +75,25 @@ class DCont():
         Create new DArray object.
     """
 
-    def __init__(self, pool=None, cont=None, path=None):
+    def __init__(self, pool=None, cont=None, path=None, open_mode='RW'):
         self._dc = DaosClient()
         self._hdl = None
         if path is None and (pool is None or cont is None):
             raise PyDError("invalid pool or container UUID",
                            -pydaos_shim.DER_INVAL)
+        open_ro = False
+        if open_mode == 'RO':
+            open_ro = True
+        elif open_mode != 'RW':
+            raise InvalidModeError('open_mode is neither "RO" or "RW"')
```

Comment:
I don't know what's best here, but we could use the built-in `ValueError` which is most similar to EINVAL

Reply:
If you see on L23 this inherits from `ValueError` so could be caught as ValueError but also allows more specific error messages and catching.